tests/v2/validation/networking/connectivity/network_policy_test.go

```diff
@@ -0,0 +1,148 @@
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+type NetworkPolicyTestSuite struct {
+	suite.Suite
+	session     *session.Session
+	client      *rancher.Client
+	project     *management.Project
+	clusterName string
+	namespace   *v1.SteveAPIObject
+	steveClient *v1.Client
+}
+
+const (
+	nodeRole = "control-plane"
+	// Ping once
+	pingCmd        = "ping -c 1"
+	successfulPing = "0% packet loss"
+)
+
+func (n *NetworkPolicyTestSuite) TearDownSuite() {
+	n.session.Cleanup()
+}
+
+func (n *NetworkPolicyTestSuite) SetupSuite() {
+	testSession := session.NewSession()
+	n.session = testSession
+
+	client, err := rancher.NewClient("", testSession)
+	require.NoError(n.T(), err)
+
+	n.client = client
+
+	clusterName := client.RancherConfig.ClusterName
+	require.NotEmpty(n.T(), clusterName, "Cluster name to install is not set")
+	n.clusterName = clusterName
+
+	cluster, err := clusters.NewClusterMeta(client, clusterName)
+	require.NoError(n.T(), err)
+
+	projectConfig := &management.Project{
+		ClusterID: cluster.ID,
+		Name:      pingPodProjectName,
+	}
+
+	createdProject, err := client.Management.Project.Create(projectConfig)
+	require.NoError(n.T(), err)
+	require.Equal(n.T(), createdProject.Name, pingPodProjectName)
+	n.project = createdProject
+
+	names := newNames()
+
+	namespaceName := names.random["namespaceName"]
+	daemonsetName := names.random["daemonsetName"]
+
+	n.T().Logf("Creating namespace with name [%v]", namespaceName)
+	n.namespace, err = namespaces.CreateNamespace(n.client, namespaceName, "{}", map[string]string{}, map[string]string{}, n.project)
+	require.NoError(n.T(), err)
+	assert.Equal(n.T(), n.namespace.Name, namespaceName)
+
+	n.steveClient, err = n.client.Steve.ProxyDownstream(n.project.ClusterID)
+	require.NoError(n.T(), err)
+
+	testContainerPodTemplate := newPodTemplateWithTestContainer()
+
+	n.T().Logf("Creating a daemonset with the test container with name [%v]", daemonsetName)
+	daemonsetTemplate := shepworkloads.NewDaemonSetTemplate(daemonsetName, n.namespace.Name, testContainerPodTemplate, true, nil)
+	createdDaemonSet, err := n.steveClient.SteveType(workloads.DaemonsetSteveType).Create(daemonsetTemplate)
+	require.NoError(n.T(), err)
+	assert.Equal(n.T(), createdDaemonSet.Name, daemonsetName)
+
+	n.T().Logf("Waiting for daemonset [%v] to have expected number of available replicas", daemonsetName)
+	err = charts.WatchAndWaitDaemonSets(n.client, n.project.ClusterID, n.namespace.Name, metav1.ListOptions{})
+	require.NoError(n.T(), err)
+}
+
+func (n *NetworkPolicyTestSuite) TestPingPods() {
+
+	// Get downstream cluster wrangler context
+	wc, err := n.client.WranglerContext.DownStreamClusterWranglerContext(n.project.ClusterID)
+	require.NoError(n.T(), err)
+
+	pods, err := wc.Core.Pod().List(n.namespace.Name, metav1.ListOptions{})
+	assert.NoError(n.T(), err)
+	assert.NotEmpty(n.T(), pods)
+
+	pod2Ip := pods.Items[1].Status.PodIP
+	pingExecCmd := pingCmd + " " + pod2Ip
```

Comment:
We should probably define these close to where they're being used just for readability.

Also, we probably want to ping all the pods associated with this daemonset, not just one of them. This way, in case the one we ping is on the node we are pinging from (it could be control-plane + worker), we ensure both inter-node connectivity as well as intra-node connectivity.

Reply:
Sure, just trying to follow the python test.

Reply:
No problem. We should definitely update to check connectivity to all the pods though to really ensure this test does what it should be doing!

Reply:
Done.